src/routes/transactions/transactions.service.ts

```diff
@@ -175,4 +185,119 @@ export class TransactionsService {
       results,
     };
   }
+
+  private getDayInMillis(timestamp: number, timezoneOffset?: string): number {
+    const date = new Date(timestamp);
+    if (timezoneOffset !== undefined) {
+      date.setUTCSeconds(parseInt(timezoneOffset) || 0);
+    }
+    return Date.UTC(date.getFullYear(), date.getMonth(), date.getDate());
+  }
+
+  private groupByDay(
+    transactions: TransactionHistory[],
```

Comment:
Why not use the domain entity?

Reply:
Because I need map first the transaction domain to `TransactionHistory`. For example each Transfer in `EthereumTransaction` is a `TransactionHistory` with timestamp and this timestamps should be considered each one to be grouped by day.

Reply:
Done https://github.com/5afe/safe-client-gateway-nest/pull/242/commits/d4188008ab148ab6a23fbd76f30d96e2ba14e495 :smiley: